sentry-ruby/spec/sentry_spec.rb

```diff
@@ -1032,4 +1032,53 @@
       end.to change { new_transport.events.count }.by(1)
     end
   end
+
+  describe "patching" do
+    context "with target and patch to prepend" do
+      let(:target_class) { Class.new }
+
+      let(:module_patch) do
+        Module.new do
+          def foo; end
+        end
+      end
+
+      before do
+        described_class.register_patch(:module_patch, module_patch, target_class)
+      end
+
+      it "does not prepend patch if not in enabled_patches" do
+        expect(target_class).not_to receive(:prepend)
+        perform_basic_setup
+        expect(target_class.ancestors).not_to include(module_patch)
+        expect(target_class.instance_methods).not_to include(:foo)
+      end
+
+      it "prepends patch if in enabled_patches" do
+        expect(target_class).to receive(:prepend).with(module_patch).and_call_original
+        perform_basic_setup  { |c| c.enabled_patches = %i(module_patch) }
+
+        expect(target_class.ancestors.first).to eq(module_patch)
+        expect(target_class.instance_methods).to include(:foo)
+      end
+    end
+
+    context "with block" do
+      let(:dbl) { double("block") }
+
+      before do
+        described_class.register_patch(:block_patch) { dbl.foo }
```

Comment:
Can we do `target_class.send(:prepend, module_patch)`, verify the result by checking `instance_methods`, and not use the double?

I think by aligning the testing styles they'll be easier to understand.